integrations/terraform-cloud/client.py

```diff
@@ -0,0 +1,144 @@
+
+                if len(workspaces) < PAGE_SIZE:
+                    break
+                page += 1
+
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error with status code : {e.response.status_code} and response text: {e.response.text}")
+        
+        except httpx.HTTPError as e:
+            logger.error(f"An error occured while fetching terraform data: {e}")
+    
+
+    async def get_paginated_runs_for_workspace(self, workspace_id: str) -> AsyncGenerator[Dict[str, Any], None]:
+        
+        try:
+            page = 1
+
+            logger.info(f"Getting runs for workspace {workspace_id} from Terraform")
+            while True:
+                run_response = await self.client.get(
+                    f"{self.api_url}/workspaces/{workspace_id}/runs?page[number]={page}&page[size]={PAGE_SIZE}"
+                )
+                run_response.raise_for_status()
+                runs_data = run_response.json()
+                runs = runs_data.get('data', [])
+                yield runs
+
+                if len(runs) < PAGE_SIZE:
+                    break
+                page += 1
+
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error with status code : {e.response.status_code} and response text: {e.response.text}")
+        
+        except httpx.HTTPError as e:
+            logger.error(f"An error occured while fetching terraform data: {e}")
+
+
+    async def get_single_run(self, run_id: str) -> Dict[str, Any]:
+        try:
+            run_url = f"{self.api_url}/runs/{run_id}"
+            response = await self.client.get(run_url)
+            response.raise_for_status()
+            return response.json().get('data', {})
+        
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error with status code : {e.response.status_code} and response text: {e.response.text}")
+        
+        except httpx.HTTPError as e:
+            logger.error(f"An error occured while fetching terraform data: {e}")
+
+
+    async def create_workspace_webhook(self, app_host: str) -> None:
+        try:
+            webhook_target_url = f"{app_host}/integration/webhook"
+
+            async for workspaces in self.get_paginated_workspaces():
+                for workspace in workspaces:
+                    notification_config_url = f"{self.api_url}/workspaces/{workspace['id']}/notification-configurations"
+
+                    existing_configs_response = await self.client.get(notification_config_url)
+                    existing_configs_response.raise_for_status()
+                    existing_configs = existing_configs_response.json().get('data', [])
+
+                    for config in existing_configs:
+                        if config['attributes']['url'] == webhook_target_url:
+                            logger.info(f"Webhook already exists for workspace {workspace['id']}")
+                            break
+
+                    else:

```

Comment:
For Loop: Iterates through each existing webhook configuration (config) for a workspace.

Condition Inside the Loop: Checks if the current config's URL matches the target webhook URL (webhook_target_url).

Break Statement: If a matching webhook URL is found, the loop is exited prematurely using a break statement, and the program logs that the webhook already exists for this workspace.

Else Block: If no matching webhook URL is found (meaning the for loop completes without hitting a break), the else block is executed. This block contains code to create a new webhook configuration.

This is because the completion of the for loop without a break indicates that no existing configuration matches the target URL, signifying the need to create a new webhook.